ccmlib/scylla_node.py

```diff
@@ -1335,6 +1336,34 @@ def upgrade(self, upgrade_to_version):
     def rollback(self, upgrade_to_version):
         self.upgrader.upgrade(upgrade_version=upgrade_to_version, recover_system_tables=True)
 
+    def watch_rest_for_alive(self, nodes, timeout=120):
+        """
+        Use the REST API to wait until this node detects that the nodes listed
+        in "nodes" become fully operational (live and no longer "joining").
+        This is similar to watch_log_for_alive but uses ScyllaDB's REST API
+        instead of the log file and waits for the node to be really useable,
+        not just "UP" (see issue #461)
+        """
+        tofind = nodes if isinstance(nodes, list) else [nodes]
+        tofind = set([node.address() for node in tofind])
+        url_live = f"http://{self.address()}:10000/gossiper/endpoint/live"
+        url_joining = f"http://{self.address()}:10000/storage_service/nodes/joining"
+        endtime = time.time() + timeout
+        while time.time() < endtime:
+            live = {}
+            response = requests.get(url=url_live)
+            if response.text:
+                live = set(response.json())
+            response = requests.get(url=url_joining)
+            if response.text:
+                live = live - set(response.json())
+            if tofind.issubset(live):
+                # This node thinks that all given nodes are alive and not
+                # "joining", we're done.
+                return
+            time.sleep(0.1)
```

Comment:
I think this sleep can be shorter

Reply:
I'm happy that from tests that take 5 minutes, we went to "0.1 second is too long" :-)
What kind of sleep did you have in mind? Say 0.05? 0.01?

In any case, note that this sleep time is not cumulative: If we slept 0.1 seconds for the benefit of one node, by the time we go check the other node, the check is likely to pass immediately, without any sleep at all. So at worst, this sleep can add 0.1 seconds to the total cluster-creation time (well, not exactly... we also do this check after adding each node, so maybe 0.1 second per node).